trace/kafka/component.go

```diff
@@ -0,0 +1,158 @@
+package kafka
+
+import (
+	"errors"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/beatlabs/patron/encoding"
+	"github.com/beatlabs/patron/encoding/json"
+	patronErrors "github.com/beatlabs/patron/errors"
+	"github.com/beatlabs/patron/log"
+	"github.com/opentracing/opentracing-go"
+)
+
+// RequiredAcks is used in Produce Requests to tell the broker how many replica acknowledgements
+// it must see before responding.
+type RequiredAcks int16
+
+const (
+	// NoResponse doesn't send any response, the TCP ACK is all you get.
+	NoResponse RequiredAcks = 0
+	// WaitForLocal waits for only the local commit to succeed before responding.
+	WaitForLocal RequiredAcks = 1
+	// WaitForAll waits for all in-sync replicas to commit before responding.
+	WaitForAll RequiredAcks = -1
+)
+
+const fieldSetMsg = "Setting property '%v' for '%v'"
+
+// Builder gathers all required and optional properties, in order
+// to construct a Kafka AsyncProducer.
+type Builder struct {
+	brokers     []string
+	cfg         *sarama.Config
+	chErr       chan error
+	tag         opentracing.Tag
+	enc         encoding.EncodeFunc
+	contentType string
+	errors      []error
+}
+
+// NewBuilder initiates the AsyncProducer builder chain.
+// The builder instantiates the component using default values for
+// EncodeFunc and Content-Type header.
+func NewBuilder(brokers []string) *Builder {
+	cfg := sarama.NewConfig()
+	cfg.Version = sarama.V0_11_0_0
+
+	return &Builder{
+		brokers:     brokers,
+		cfg:         cfg,
+		chErr:       make(chan error),
+		tag:         opentracing.Tag{Key: "type", Value: "async"},
+		enc:         json.Encode,
+		contentType: json.Type,
+		errors:      []error{},
+	}
+}
+
+// WithTimeout sets the dial timeout for the AsyncProducer.
+func (ab *Builder) WithTimeout(dial time.Duration) *Builder {
+	if dial <= 0*time.Second {
```

Comment:
i dont think it's needed to add the time.Second part. 'dial <= 0' should work the same.